### PR TITLE
fix CI for initial SDK codegen PR

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

1. Removed `EnableBatchRelease` from ci.yml, due to: https://github.com/Azure/azure-sdk-for-java/commit/14b17fac7456ba84df914cc73e970802e6342fa4
2. Added `Features Added` section to initial changelog.md, similar to https://github.com/Azure/azure-sdk-for-java/blob/f1904e279c2f34ef7eb8339c5761022e7d01e6a6/sdk/devcenter/azure-developer-devcenter/CHANGELOG.md?plain=1#L117-L122
This is Due to new CI check on changelog.md: https://github.com/Azure/azure-sdk-for-java/pull/40946#issuecomment-2199717150
This affects both dpg/mpg, swagger/tsp. 
3. Fix https://github.com/Azure/autorest.java/issues/2846
4. example: https://github.com/Azure/azure-sdk-for-java/pull/40946
5. Tested locally.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/mgmt/automation/utils.py [hunk 1]
```diff
@@ -175,6 +178,19 @@ def update_service_ci_and_pom(sdk_root: str, service: str, group: str, module: s
...
 
+    if os.path.exists(changelog_file):
+        with open(changelog_file, "r") as fin:
+            changelog_str = fin.read()
+    logging.info("[CHANGELOG][Process] Adding initial section to changelog.md..")
+    initial_section = CHANGELOG_INITIAL_SECTION_FORMAT.format(artifact_id=module)
+    if initial_section not in changelog_str:
+        changelog_str += initial_section
+        with open(changelog_file, "w") as fout:
+            fout.write(changelog_str)
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/utils.py:189]: Is this a re-write of the whole file? What about version etc.? Should it be `w+`?

File: eng/mgmt/automation/utils.py [hunk 2]
```diff
@@ -175,6 +178,19 @@ def update_service_ci_and_pom(sdk_root: str, service: str, group: str, module: s
             fout.write(pom_xml)
         logging.info("[POM][Success] Write to pom.xml")
 
+    if os.path.exists(changelog_file):
+        with open(changelog_file, "r") as fin:
+            changelog_str = fin.read()
+    logging.info("[CHANGELOG][Process] Adding initial section to changelog.md..")
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/utils.py:184]: nit, too many dot

File: eng/mgmt/automation/utils.py [hunk 3]
```diff
@@ -175,6 +178,19 @@ def update_service_ci_and_pom(sdk_root: str, service: str, group: str, module: s
...
+    initial_section = CHANGELOG_INITIAL_SECTION_FORMAT.format(artifact_id=module)
+    if initial_section not in changelog_str:
+        changelog_str += initial_section
+        with open(changelog_file, "w") as fout:
+            fout.write(changelog_str)
+        logging.info("[CHANGLOG][Success] Write to changelog.md.")
+    else:
+        logging.warning("[CHANGELOG] changelog.md already contains necessary sections. Will not add initial sections.")
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/utils.py:192]: This seems only work for the 1st release.

Another candidate place is in `update_changelog` func.

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/utils.py:192]: BTW, will it be invoked for SDK with existing releases, but without this `- Initial release for the {artifact_id} Java SDK.`?

Comment (XiaofeiCao) [reply eng/mgmt/automation/utils.py:192]: Yeah, this PR was intended only for 1st release.

Yes, there could be cases where there's no new features/breaking changes, though rare. 
Though I'm wondering what we should add to changelog content in this case. 
How about 
```md
### Other Changes

#### Dependency Updates

- Updated service `api-version`. 
```
?

Comment (XiaofeiCao) [reply eng/mgmt/automation/utils.py:192]: > BTW, will it be invoked for SDK with existing releases, but without this `- Initial release for the {artifact_id} Java SDK.`?

No, this method is only called when `require_sdk_integration` is true, which is 1st release.

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/utils.py:192]: We can ignore this for now.

Comment (XiaofeiCao) [reply eng/mgmt/automation/utils.py:192]: > No, this method is only called when `require_sdk_integration` is true, which is 1st release.

Emm, found one exception in generating data-plane SDK if no readme:
https://github.com/Azure/azure-sdk-for-java/blob/20828f55b82becc3691550f90a9220fe77df9552/eng/mgmt/automation/generate_data.py#L346-L349

This is the only exception. I guess this still means 1st release?

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/utils.py:192]: OK. DPG likely use a slightly different meaning of `require_sdk_integration`

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/utils.py:192]: Then it should be OK, if only called for codegen of new lib.

Comment (XiaofeiCao) [reply eng/mgmt/automation/utils.py]: Updated the method's name to better reflex what it's meant to do.

File: eng/mgmt/automation/utils.py (hunk 2 above)
Comment (XiaofeiCao) [reply eng/mgmt/automation/utils.py]: Updated.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**